snyk/models.py

```diff
@@ -176,6 +176,15 @@ def _test(self, path, contents=None):
                 "encoding": "base64",
                 "files": {"target": {"contents": encoded}},
             }
+
+            # Some test methods carry a second file, often a lock file
+            if additional:
```

Comment:
Python is [EAFP than LBYL ](https://www.geeksforgeeks.org/eafp-principle-in-python/) therefore a Pythonic way of writing this code block would look like:
```python
try:
    read = getattr(additional, "read", None)
    additional = additional.read()
    encoded = base64.b64encode(additional.encode()).decode()
    post_body["files"]["additional"] = {"contents": encoded}
except:
    pass
```

I'm aware of the fact that the code block at the top of this method is LBYL and therefore this one is too, but worth noting IMHO

Reply:
I think I refer the consistency vs having two stanzas with different styles. Not against someone refactoring the whole method mind. It requires some additional type hints I think.

Reply:
You have a point there. let's take this one separately as part of a refactor (happy to take it)